library/src/main/java/org/jellyfin/apiclient/interaction/connectionmanager/ValidateAuthenticationResponse.java

```diff
@@ -36,14 +33,19 @@ public void onResponse(String jsonResponse) {
         SystemInfo obj = jsonSerializer.DeserializeFromString(jsonResponse, SystemInfo.class);
         server.ImportInfo(obj);
 
-        if (!tangible.DotNetToJavaStringHelper.isNullOrEmpty(server.getUserId()))
-        {
-            request.setUrl(url + "/mediabrowser/users/" + server.getUserId() + "?format=json");
-
-            httpClient.Send(request, new ValidateAuthenticationInnerResponse(connectionManager, jsonSerializer, server, response));
-        } else {
-            response.onResponse();
+        if (!tangible.DotNetToJavaStringHelper.isNullOrEmpty(server.getUserId()) &&
+                !tangible.DotNetToJavaStringHelper.isNullOrEmpty(server.getId())) {
+            ApiClient client = connectionManager.GetApiClient(server.getId());
+            if (client != null) {
+                String apiUrl = client.GetApiUrl("Users/" + server.getUserId());
+                apiUrl = client.AddDataFormat(apiUrl);
+                request.setUrl(apiUrl);
+
+                httpClient.Send(request, new ValidateAuthenticationInnerResponse(connectionManager, jsonSerializer, server, response));
+                return;
+            }
         }
+        response.onResponse();
```

Comment:
Shouldn't this stay in an `else` block? The inner response may not have been triggered yet.

Reply:
The `return` above should be used in the event that this should not be called. Otherwise we would need to handle the case where `client` is `null` also. I thought this approach would be cleaner.

I tested this in Android TV and it seems to be working as expected there for what it's worth.